### ISSUE TITLE
unable to create new assignment in dev (to test new Admin UI functionality)

### ISSUE THREAD

**Issue by ShannonRice** [Harvard-ATG/annotationsx#68]
[ANNO-127 in Jira]

I am unable to create a new assignment in Dev. (to test out the new Admin UI)
Steps to recreate: 
1) Go to dialog box to create a new assignment. 
2) Select "create a new source" 
3) Fill out fields of dialogue box that comes up.
At this point, a white box appears, when it is closed, the following Errors appear (See attachment)
Attachments

Screen Shot 2016-07-13 at 3.50.45 PM.png
339 kB5 minutes ago
![screen shot 2016-07-13 at 3 50 45 pm](https://cloud.githubusercontent.com/assets/2422354/16817810/9b39135c-4912-11e6-933a-0df70739aab6.png)

**Comment by arthurian** [comment]
@ShannonRice I'm having trouble duplicating this error exactly as shown in your screenshot. When I submit the form to create a new source, it appears to submit OK, although the dialog window just goes blank and never closes. Looking in the developer console, I see a Javascript-related error:

*/static/vendors/development/ExpandedRelatedObjectLookups.js:93 Uncaught TypeError: window.add_source_row is not a function*

I'll do some more investigating...

**Comment by arthurian** [comment]
@ShannonRice  ... so after more investigating, I was able to duplicate the error: it occurs if you try to save an assignment without *any* sources, whether that's a source you've just created, or a source you've selected from the existing repository. So I see two things that need to be fixed here:

1. We need to fix the popup dialog so that when you click "save" on the new source, it properly adds the new source to the assignment in the main browser window (this is not happening to the error I posted in my previous comment).
2. We need to fix the assignment submission handler so that it's possible to either (a) create an assignment without any sources attached or (b) displays a friendly warning message to the user that they are required to attach at least one source in order to create an assignment. What was the previous/existing behavior on production?

@lduarte1991 Are you aware of this issue on your end as well?
